### PR TITLE
Display cart price according to select country from the location selector (price OR price + shipping price)

### PR DESCRIPTION
**Pull request by cis-shubham-t**
- Display the cart total according to the selected country from the location selector.

- display the section shipping price option at the checkout process item listing for the first process, if the country is selected from the location.

### CODE REVIEW

**Comment by cis-shubham-t** [comment]
@AllienWorks 
The dot (.) issue fixed now can please have a look :)  

fixed

**Comment by cis-shubham-t** [comment]
@AllienWorks 
@HenkSwardt 

I've added the mechanism to remove the focus from the input box when the user selects an option from the available options list.
That's resolved our issue related with the location selector and other search select box.

Please have a look :)

**Comment by AllienWorks** [comment]
> There's a small issue of Shipping price in Cart - it's not displaying the . in price.

Yes, this has been resolved. Good job!

> I've added the mechanism to remove the focus from the input box when the user selects an option from the available options list.

Tested & works as needed.

**Still needs code review.**

File: src/app/market/listings/listings.component.ts [hunk 1]
```diff
@@ -79,7 +79,11 @@ export class ListingsComponent implements OnInit, OnDestroy {
     private countryList: CountryListService
   ) {
     console.warn('overview created');
-    this.selectedCountry = this.listingService.cache.selectedCountry;
+    if (this.listingService.cache.selectedCountry) {
```
Comment (kewde) [inline src/app/market/listings/listings.component.ts]: Why just not make it a type country in the post-listings cache?

File: src/app/core-ui/material/mat-select-search/mat-select-search.component.ts [hunk 2]
```diff
@@ -59,15 +68,20 @@ export class MatSelectSearchComponent implements OnInit, OnChanges {
     return this.options.filter(option => option[this.showValueOf].toLowerCase().includes(filterValue));
   }
 
-  public _onBlur($event: any): void {
-    const val = $event.target.value;
+  public _onBlur(): void {
```
Comment (kewde) [inline src/app/core-ui/material/mat-select-search/mat-select-search.component.ts]: Why was $event not working?

Comment (cis-shubham-t) [reply src/app/core-ui/material/mat-select-search/mat-select-search.component.ts]: yes, that's working fine.
But in the same component, we have the ``Viewchild`` (native element) ref of the <input-field>.
that's the reason I'm replaced ;

``` $event.target = this.textInput.nativeElement  ```

**Comment by AllienWorks** [comment]
LGTM

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 3]
```diff
@@ -42,9 +42,9 @@
         </td>
       </tr>
 
-      <tr *ngIf="listingServiceCache?.selectedCountry">
+      <tr *ngIf="listingServiceCache?.selectedCountry?.iso">
```
[the remark this answers is not in the document]
Comment (cis-shubham-t) [reply src/app/market/listings/preview-listing/preview-listing.component.html]: Yes, working in that case as well.

Its just to added for make condition more specfic.

**Pull request merged**